ipatests/pytest_ipa/integration/host_ipaplatform.py

```diff
@@ -0,0 +1,238 @@
+                    import json
+
+                    from ipaplatform.paths import paths
+
+
+                    remote_paths = {}
+                    for name in sorted(dir(paths)):
+                        if name.startswith("_"):
+                            continue
+
+                        value = getattr(paths, name)
+                        try:
+                            json.dumps(value)
+                        except TypeError:
+                            continue
+
+                        remote_paths[name] = value
+
+                    json_data = json.dumps(remote_paths)
+                    json_base64 = base64.b64encode(
+                        json_data.encode("utf-8")
+                    ).decode("ascii")
+                    print(json_base64)
+                """
+            )
+            cmd = ["python3", "-c", code]
+            res = self.run_command(cmd, log_stdout=False)
+            json_data = base64.b64decode(res.stdout_bytes).decode("utf-8")
+            self._paths = HostPlatformPaths(**json.loads(json_data))
+
+        return self._paths
+
+    @property
+    def constants(self):
+        if self._constants is None:
+            code = textwrap.dedent(
+                """\
+                    import base64
+                    import json
+                    from ipaplatform.constants import constants
+
+
+                    remote_constants = {}
+                    for name in sorted(dir(constants)):
+                        if name.startswith("_"):
+                            continue
+
+                        value = getattr(constants, name)
+                        try:
+                            json.dumps(value)
+                        except TypeError:
+                            continue
+
+                        remote_constants[name] = value
+
+                    json_data = json.dumps(remote_constants)
+                    json_base64 = base64.b64encode(
+                        json_data.encode("utf-8")
+                    ).decode("ascii")
+                    print(json_base64)
+                """
+            )
+            cmd = ["python3", "-c", code]
+            res = self.run_command(cmd, log_stdout=False)
+            json_data = base64.b64decode(res.stdout_bytes).decode("utf-8")
+            self._constants = HostPlatformConstants(**json.loads(json_data))
+
+        return self._constants
+
+    @property
+    def osinfo(self):
+        if self._osinfo is None:
+            code = textwrap.dedent(
+                """\
+                    import base64
+                    import json
+
+                    from ipaplatform.osinfo import osinfo
+
+
+                    remote_osinfo = {}
+                    for name in sorted(dir(osinfo)):
+                        if name.startswith("_"):
+                            continue
+
+                        value = getattr(osinfo, name)
+                        try:
+                            json.dumps(value)
```

Comment:
For the version_number attribute (which is a tuple), this call to json.dumps + json.loads transforms the tuple into a list:
```
>>> import json
>>> from ipaplatform.osinfo import osinfo
>>> value = getattr(osinfo, 'version_number')
>>> value
(33,)
>>> json.loads(json.dumps(value))
[33]
```
and that later creates an issue in the test:
```    def test_smb_service_s4u2self(self):
        """Test S4U2Self operation by IPA service
           against both AD and IPA users
        """
        master_os = self.master.ipaplatform.osinfo
>       if master_os.id == "fedora" and master_os.version_number <= (31,):
E       TypeError: '<=' not supported between instances of 'list' and 'tuple'

test_integration/test_smb.py:338: TypeError
```